src/main/java/me/kixstar/dungeons/dungeons/ChestUtils.java

```diff
@@ -1,52 +1,53 @@
 package me.kixstar.dungeons.dungeons;
 
 import org.bukkit.block.Chest;
-import org.bukkit.entity.Item;
-import org.bukkit.entity.Player;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.ItemStack;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
 
 public class ChestUtils {
 
     private static final Random random = new Random();
+
     /**
      * distributeItems distributes a list of items into different chests, randomly
+     *
      * @param chests
      * @param items
      */
     static void distributeItems(List<Chest> chests, List<ItemStack> items) {
-        if (chests != null && items != null && !(items.isEmpty())) {
-            for (Chest chest : chests) {
-                List<ItemStack> itemStacks = null;
-                for (int i = 0; i < random.nextInt(items.size()); i++) {
-                    itemStacks.add(items.get(i));
-                    items.remove(i);
-                }
-                distributeItems(chest.getInventory(), itemStacks);
-            }
-        } else {
+        if (chests == null || items == null || items.isEmpty()) {
```

Comment:
Talked to Raptor, we'd prefer to throw an Exception if chests and items are null, but just return if the list of items are empty

Reply:
What kind of exception do you want it to throw?

Reply:
IllegalArgument should do the trick